pulsar-broker/src/main/java/org/apache/pulsar/broker/transaction/buffer/impl/SingleSnapshotAbortedTxnProcessorImpl.java

```diff
@@ -87,7 +90,16 @@ public CompletableFuture<PositionImpl> recoverFromSnapshot() {
                     PositionImpl startReadCursorPosition = null;
                     try {
                         while (reader.hasMoreEvents()) {
-                            Message<TransactionBufferSnapshot> message = reader.readNext();
+                            Message<TransactionBufferSnapshot> message = reader.readNext(2, TimeUnit.SECONDS);
+                            if (message == null){
+                                String warnLog = String.format("[%s] When reading from topic %s,the latest message has"
+                                                + " been deleted by compaction-task or trim ledger.",
+                                        topic.getName(),
+                                        SystemTopicNames.TRANSACTION_BUFFER_SNAPSHOT);
+                                log.warn(warnLog);
```

Comment:
Maybe there should use the debug log level because this is a normal case.

Reply:
Hi @liangyepianzhou 

I used the [simper logic](https://github.com/apache/pulsar/pull/18833#discussion_r1045360674) suggested by @congbobo184, could you retake a look?